### PR TITLE
[PORT]  Ashwalker breathing and oxygen toxcicity

### PR DESCRIPTION
**Pull request by Flakkyboo**
## About The Pull Request

Ported from https://github.com/Citadel-Station-13/Citadel-Station-13/pull/8159

Currently Ashwalkers do not breathe, they do not care about atmos in any fashion this change aims to fix that by giving them lungs. Ashwalker lungs are only suited to the low pressure environment of lavaland and will begin to suffocate if they go into areas with atmosphere suited for humans.

This also makes it so if you are in an environment with too high an oxygen concentration you will begin to get dizzy and take suffocation damage gradually until you leave get some better air.

It is now compatible with the randomized lavaland atmosphere. (many thanks to St0rm for the assist)

## Why It's Good For The Game
This is step one in a process to make ashwalkers more than a card to murderbone and more of a living species on a planet.

While you can still do the unga bunga limitations in atmosphere make it so going up to the station for long periods of time is more of a danger. 

Makes high oxygen environments more dangerous for everyone.

## Changelog
:cl:
add: New Ashwalker lungs "Ash Lungs"
add: Ashwalkers can only breath lavaland atmosphere.
add: Code to allow for suffocation if breathing in too much oxygen.
add: Compatibility with randomized lavaland atmosphere.
add: New sprite for Ashwalker lungs.
add: Code that allows Ashwalkers to process BZ moles very quickly in the form of a minuscule heal
tweak: Removed unneeded code related to plasmamen suffocating from too much oxygen as they already suffocate without plasma.
tweak: Added a maximum oxygen threshold for cybernetic lungs.
tweak: Adjusted the randomized lavaland atmos base values to prevent situations where o2 or n2
goes above ashwalker maximums.
Fix: Lava tiles around necropolis tendril now use lavaland atmos 
Fix: Fixed conflicts with beepeople lungs
/:cl:

### CODE REVIEW

**Comment by St0rmC4st3r** [comment]
Don't revert the random lavaland atmos code. Just make them either immune to secondary gasses or make something like BZ beneficial to them.

**Review by ike709 — CHANGES_REQUESTED** [review]
> del: reverted code related to randomized lavaland atmosphere

This PR isn't getting merged as long as random atmos is removed. Have ashwalkers breathe whatever the gasses of the round are.

File: code/game/turfs/open.dm [hunk 1]
```diff
@@ -87,7 +87,9 @@
 		icon_state = "necro[rand(2,3)]"
 
 /turf/open/indestructible/necropolis/air
-	initial_gas_mix = OPENTURF_DEFAULT_ATMOS
+	//initial_gas_mix = OPENTURF_DEFAULT_ATMOS
```
Comment (ike709) [inline code/game/turfs/open.dm]: Don't comment out code, just remove it.

File: code/game/turfs/open.dm [hunk 2]
```diff
@@ -97,7 +99,8 @@
 	initial_gas_mix = LAVALAND_DEFAULT_ATMOS
 
 /turf/open/indestructible/boss/air
-	initial_gas_mix = OPENTURF_DEFAULT_ATMOS
+	//initial_gas_mix = OPENTURF_DEFAULT_ATMOS
```
Comment (ike709) [inline code/game/turfs/open.dm]: Don't comment out code, just remove it.

File: code/game/turfs/simulated/lava.dm [hunk 3]
```diff
@@ -168,6 +168,8 @@
 	smooth = SMOOTH_MORE | SMOOTH_BORDER
 	canSmoothWith = list(/turf/open/lava/smooth)
 
+
+
```
Comment (ike709) [inline code/game/turfs/simulated/lava.dm]: Random whitespace addition?

File: code/modules/mob/living/carbon/life.dm [hunk 4]
```diff
@@ -167,8 +168,21 @@
...
+			adjustOxyLoss(3)
+			if(world.time - o2overloadtime > 300)
+				adjustOxyLoss(8)
+		if(prob(20))
+			emote("cough")
+		throw_alert("too_much_oxy", /obj/screen/alert/too_much_oxy)
+		SEND_SIGNAL(src, COMSIG_ADD_MOOD_EVENT, "suffocation", /datum/mood_event/suffocation)
+
+
 	if(O2_partialpressure < safe_oxy_min) //Not enough oxygen
```
Comment (ike709) [inline code/modules/mob/living/carbon/life.dm]:
```suggestion
	else if(O2_partialpressure < safe_oxy_min) //Not enough oxygen
```

File: code/modules/surgery/organs/lungs.dm [hunk 5]
```diff
@@ -106,11 +107,25 @@
 
 	//Too much oxygen! //Yes, some species may not like it.
 	if(safe_oxygen_max)
-		if(O2_pp > safe_oxygen_max)
+		if((O2_pp > safe_oxygen_max) && safe_oxygen_max ==0)
```
Comment (ike709) [inline code/modules/surgery/organs/lungs.dm]: This check will never pass.

File: code/modules/surgery/organs/lungs.dm [hunk 6]
```diff
@@ -106,11 +107,25 @@
...
 			var/ratio = (breath_gases[/datum/gas/oxygen][MOLES]/safe_oxygen_max) * 10
 			H.apply_damage_type(CLAMP(ratio, oxy_breath_dam_min, oxy_breath_dam_max), oxy_damage_type)
 			H.throw_alert("too_much_oxy", /obj/screen/alert/too_much_oxy)
+
+		else if((O2_pp > safe_oxygen_max) && !(safe_oxygen_max == 0))
```
Comment (ike709) [inline code/modules/surgery/organs/lungs.dm]: What

File: code/modules/surgery/organs/lungs.dm [hunk 7]
```diff
@@ -444,8 +464,27 @@
...
 	safe_toxins_max = 20
 	safe_co2_max = 20
+	safe_oxygen_max = 250
 	maxHealth = 2 * STANDARD_ORGAN_THRESHOLD
 
 	cold_level_1_threshold = 200
 	cold_level_2_threshold = 140
 	cold_level_3_threshold = 100
+
+obj/item/organ/lungs/ashwalker
```
Comment (ike709) [inline code/modules/surgery/organs/lungs.dm]:
```suggestion
/obj/item/organ/lungs/ashwalker
```

File: code/game/turfs/simulated/lava.dm (hunk 3 above)
Comment (Flakkyboo) [reply code/game/turfs/simulated/lava.dm]: ah that was what was left over from my hacky fix of the nest atmos before i used fastdmm.
forgot to get rid of the extra lines after i got rid of the commented out initial atmos.

thanks for looking over this ill make those changes now

File: code/modules/surgery/organs/lungs.dm (hunk 5 above)
Comment (Flakkyboo) [reply code/modules/surgery/organs/lungs.dm]: I ended up removing these checks "&& safe_oxygen_max ==0)" as every race that wasnt plasmamen it worked with and plasmamen already had a hard time with oxygen and didnt need to take more suffocation damage

File: code/modules/mob/living/carbon/life.dm (hunk 4 above)
Comment (Flakkyboo) [reply code/modules/mob/living/carbon/life.dm]: applied these changes

**Comment by ike709** [comment]
After careful consideration and previous attempts to make this code functional, we've determined that this PR will not be acceptable without considerable effort from maintainers.

We recommend smaller PRs in the future until you have a better understanding of coding.

**Pull request closed without merging**